explainaboard/processors/kg_link_tail_prediction.py

```diff
@@ -283,3 +315,6 @@ def _get_true_label(self, data_point: dict):
 
     def _get_predicted_label(self, data_point: dict):
         return data_point["predictions"]
+
+    def _get_rank_data(self, data_point: dict):
+        return data_point.get("true_rank", None)
```

Comment:
Suggestion: if `true_rank` is not included in the output, could we try to get it from the system predictions before returning None? This would make it more flexible to be able to handle when ranks are not passed in, such as when the predictions always contain the true answer.

One issue with requiring `true_rank` to be provided is that it prevents us from being able to use blind test sets.

Reply:
In the case of blind test sets, would it then be reasonable to assume that the system outputs will contain the entire list of entities as predictions (instead of just the top-k)? Otherwise, there would be no way to calculate the correct MR and MRR. (Hits@k would still be accurate as long as the predictions contain at least k elements.) Thoughts?

Reply:
Yes, for now we could throw an error if neither of the two conditions are satisfied. In the future (or even in this PR), we might also add the ability to specify a "default_rank" for all things that were not included. Then the condition would be:

```
if true rank exists:
  return true rank
elif true_label in n-best list:
  return rank of true_label in n-best list
elif default_rank is not None:
  return default_rank
else:
  throw error
```